flow configuration examples (#1)
banzaicloud/one-eye-docs: docs/logging-extensions/_index.md

### Configuration options
| Variable Name | Type | Required | Default | Description |
|---|---|---|---|---|
| controlNamespace | string | Yes | - | The resources of Eventtailer will be placed into this namespace<br> |
| positionVolume | volume.KubernetesVolume | No | - | Volume definition for tracking fluentbit file positions (optional)<br> |
| workloadMetaOverrides | *types.MetaBase | No | - | Override metadata of the created resources<br> |
| workloadOverrides | *types.PodSpecBase | No | - | Override podSpec fields for the given statefulset<br> |
| containerOverrides | *types.ContainerBase | No | - | Override container fields for the given statefulset<br> |

> ### Example: Configure logging Flow to listen Eventtailer

Reviewer: ```suggestion
### Example: Configure logging Flow to route logs from an event tailer
```

> The following example uses the flow's match term to listen the previously created "sample" Eventtailer's log. 

Reviewer: ```suggestion
The following example configures a flow to route the previously created `sample-eventtailer` EventTailer's log. 
```

```bash
kubectl apply -f - <<EOF
apiVersion: logging.banzaicloud.io/v1beta1
kind: Flow
metadata:
  name: eventtailer-flow
  namespace: default
spec:
  filters:
  - tag_normaliser: {}

>   - stdout: {}

Reviewer: ```suggestion
```

  match:
  # keeps data matching to label, the rest of the data will be discarded by this flow implicitly
  - select:
      labels:

>         app.kubernetes.io/name: sample-event-tailer

Reviewer: ```suggestion
        app.kubernetes.io/name: sample-eventtailer
```

Author: // EventTailer configuration
var EventTailer = EventTailerConfig{
	TailerAffix:           "event-tailer",
	ConfigurationFileName: "config.json",
	PositionVolumeName:    "event-tailer-position",
}

Author: the label name will be "sample-event-tailer" with the current configuration

  outputRefs:

>     - null-output

Reviewer: ```suggestion
    - sample-output
```

| path | string | No | - | Path to the loggable file<br> |
| disabled | bool | No | - | Disable tailing the file<br> |

> ### Example: Configure logging Flow to listen Hosttailer

Reviewer: ```suggestion
### Example: Configure logging Flow to route host logs
```

> The following example uses the flow's match term to listen the previously created "sample" Hosttailer's log. 

Reviewer: ```suggestion
The following example uses the flow's match term to route the previously created "sample" HostTailer's log. 
```

```bash
kubectl apply -f - <<EOF
apiVersion: logging.banzaicloud.io/v1beta1
kind: Flow
metadata:
  name: hosttailer-flow
  namespace: default
spec:
  filters:
  - tag_normaliser: {}

>   - stdout: {}

Reviewer: ```suggestion
```

  # keeps data matching to label, the rest of the data will be discarded by this flow implicitly
  match:
  - select:
      labels: 

>         app.kubernetes.io/name: sample-tailer

Reviewer: ```suggestion
        app.kubernetes.io/name: sample
```

      # there might be a need to match on container name too (in case of multiple containers)
      container_names:

>         - sample-tailer-nginx-access

Reviewer: ```suggestion
        - nginx-access
```

  outputRefs:

>     - null-output

Reviewer: ```suggestion
    - sample-output
```

EOF
```

> ### Example: configuration Kubernetes host tailer with multiple tailers

Reviewer: ```suggestion
### Example: configure multiple host tailers
```

```bash
kubectl apply -f - <<EOF
apiVersion: logging-extensions.banzaicloud.io/v1alpha1
kind: HostTailer
metadata:
  name: multi-sample
spec:

>   # resources will be placed in default namespace
>   controlNamespace: default
>   # recreate resources when it's unable to change a field due to immutable status
>   enableRecreateWorkloadOnImmutableFieldChange: true

Reviewer: ```suggestion
```

  # list of File tailers
  fileTailers:
    - name: nginx-access
      path: /var/log/nginx/access.log

>       disabled: false

Reviewer: ```suggestion
```

    - name: nginx-error
      path: /var/log/nginx/error.log

>       # currently unwanted
>       disabled: true

Reviewer: ```suggestion
```

  # list of Systemd tailers
  systemdTailers:
    - name: my-systemd-tailer

>       disabled: false

Reviewer: ```suggestion
```

      maxEntries: 100
      systemdFilter: kubelet.service
EOF
```

### Configuration options

| Variable Name | Type | Required | Default | Description |
|---|---|---|---|---|
| fileTailers | []FileTailer | No | - | List of file tailers<br> |
| systemdTailers | []SystemdTailer | No | - | List of systemd tailers<br> |
| enableRecreateWorkloadOnImmutableFieldChange | bool | No | - | EnableRecreateWorkloadOnImmutableFieldChange enables the operator to recreate the<br>fluentbit daemonset and the fluentd statefulset (and possibly other resource in the future)<br>in case there is a change in an immutable field<br>that otherwise couldn't be managed with a simple update.<br> |
| controlNamespace | string | Yes | - | The resources of HostTailer will be placed into this namespace<br> |
| workloadMetaOverrides | *types.MetaBase | No | - | Override metadata of the created resources<br> |
| workloadOverrides | *types.PodSpecBase | No | - | Override podSpec fields for the given daemonset<br> |

> | containerOverrides | *types.ContainerBase | No | - | Override container fields for the given daemonset<br> |

Reviewer: ```suggestion
```

> ### Example: Configure logging Flow with explicit exclusion

Reviewer: we don't need this example, log routing is discussed in detail in the logging-operator docs
